envoy/http/alternate_protocols_cache.h

```diff
@@ -128,9 +142,18 @@ class AlternateProtocolsCache {
    * @return the number if entries in the map.
    */
   virtual size_t size() const PURE;
+
+  /**
+   * @param origin The origin to get HTTP/3 status tracker for.
+   * @return the existing status tracker or creating a new one if the orgin has
+   * corresponding alternate protocols in the cache. Otherwise, nullopt.
+   */
+  virtual OptRef<AlternateProtocolsCache::Http3StatusTracker>
+  getOrCreateHttp3StatusTracker(const Origin& origin) PURE;
```

Comment:
I would have expected this it create a tracker if one does not yet exist. (Particularly considering the name). I would have thought this would create an entry if one did not yet exist. Would that work?

Reply:
We don't automatically create an entry for get|setSrtt() right now and set|findAlternatives() seems to be the only interfaces to change the life time of an entry. So I'm not sure if adding an entry without alternate protocols populated would simplify things or not.

Reply:
I think we want some interfaces which always give us an h3 tracker if the origin has a corresponding entry in the cache. If the origin is not cached, the interfaces don't have to give an h3 tracker.

Reply:
so for DFP / auto_http3 we only do HTTP/3 if we have an alt-svc entry, so while setSrtt "only works if there's already an entry" there's always an entry.
I think we should move wint he direction of having an origin -> info map, where the info contains 
quic status (alt svc, broken/confirmed etc, cached inched info like sRTT) and HTTP/2 status, and we can add them in any order.  And we rename the cache to something akin to what Chrome does ;-)
I don't have strong feelings about what parts go in this PR - will defer to Ryan on that

Reply:
Okay, if that's the direction the cache is moving towards, for h3 tracker, we can make this getOrCreateHttp3StatusTracker() also create a cache entry if there is none. In that case, the entry will only has h3 status tracker but doesn't have the alt-svc string. And any following setAlternatives() should reuse the same entry. Does this sound right?

Reply:
Yes, that sounds right to me! Thanks, Dan!